File: data/src/commonMain/kotlin/com/example/myapplication/rawresponses/LinksJSON.kt
package com.example.myapplication.rawresponses

import kotlinx.serialization.Serializable

@Serializable
data class LinksJSON(
    val self: HrefJSON,
    val tournamentEntity: HrefJSON? = null,
    val gameEntity: HrefJSON? = null,
    val modeEntity: HrefJSON? = null,
    val matchEntity: HrefJSON? = null,
    val registrationEntity: HrefJSON? = null,
    val userEntity: HrefJSON? = null,
    val profile: HrefJSON? = null,
    val search: HrefJSON? = null,
    val tournament: HrefJSON? = null,
    val game: HrefJSON? = null,
    val mode: HrefJSON? = null,
    val match: HrefJSON? = null,
    val registration: HrefJSON? = null,
    val user: HrefJSON? = null,
    val admin: HrefJSON? = null
)

File: web-client/src/app/data/rawresponses/links-json.ts
import {HrefJSON} from './href-json';

export interface LinksJSON {
  readonly self: HrefJSON;
  readonly tournamentEntity?: HrefJSON;
  readonly gameEntity?: HrefJSON;
  readonly modeEntity?: HrefJSON;
  readonly registrationEntity?: HrefJSON;
  readonly userEntity?: HrefJSON;
  readonly profile?: HrefJSON;
  readonly search?: HrefJSON;
  readonly tournament?: HrefJSON;
  readonly game?: HrefJSON;
  readonly mode?: HrefJSON;
  readonly registration?: HrefJSON;
  readonly user?: HrefJSON;
  readonly admin?: HrefJSON;
}
